File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        head = curr = None
        while list1 or list2:
            if(((list1 and list2) and (list1.val <= list2.val)) or (list1 and not list2)):
                if not curr:
                    head = curr = list1
                else:
                    curr.next = list1
                    curr = curr.next
                list1 = list1.next                
            else:
                if not curr:
                    head = curr = list2
                else:
                    curr.next = list2
                    curr = curr.next
                list2 = list2.next
        return head

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size = 0;
        ListNode curr = head;
        while (curr != null) {
            size++;
            curr = curr.next;
        }

        int nth = size - n;
        curr = head;
        if(nth == 0) {
            return head.next;
        }
        while (nth > 1) {
            nth--;
            curr = curr.next;
        }

        curr.next = curr.next.next;

        return head;
    }
}
